src/commoncode/fetch.py

```
@@ -39,6 +39,11 @@
 # logging.basicConfig(level=logging.DEBUG, stream=sys.stdout)
 # logger.setLevel(logging.DEBUG)
 
+def test_download(url):
+    if ping_url(url) == True:
+        download_url(url, file_name=None, verify=True, timeout=10)
+    else:
+        print('Failed to start download for', url)
```

You should raise an exception here instead